### PR TITLE
Move tests to actions

### PR DESCRIPTION
**Pull request by benjeffery**
As travis is changing, move to actions like tskit has.

Also add clang-format to pre-commit and dependabot to requirements.

### CODE REVIEW

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/msprime-docs/1284/)

**Comment by benjeffery** [comment]
@jeromekelleher This is now ready to go. Windows is a mess, there must be a better way, but I give up!

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/msprime-docs/PR-1284/)

File: docs/development.rst [hunk 1]
```diff
@@ -52,23 +52,19 @@ Quickstart
...
 ****************************
 
-Three different continuous integration providers are used, which run different
+Two different continuous integration providers are used, which run different
 combinations of tests on different platforms:
 
-1. A `Github action <https://help.github.com/en/actions>`_ runs `pre-commit
-   <https://pre-commit.com/>`_ to run a variety of code style and quality checks.
+1. `Github actions <https://help.github.com/en/actions>`_ run `pre-commit
+   <https://pre-commit.com/>`_ run a variety of code style and quality checks along
```
Comment (jeromekelleher) [inline docs/development.rst:59]: This isn't quite parsing as english "run pre-commit run"

**Review by jeromekelleher — APPROVED** [review]
LGTM, minor point on docs.

You gotta do what you gotta do when it comes to Windows.

File: docs/development.rst (hunk 1 above)
Comment (benjeffery) [reply docs/development.rst]: Forest gump wrote that bit. Fixed!

**Comment by benjeffery** [comment]
@jeromekelleher Fixed up - appveyor is complaining so you will need to deactivate it.

File: docs/development.rst (hunk 1 above)
Comment (jeromekelleher) [reply docs/development.rst]: Haha, good one!

**Comment by jeromekelleher** [comment]
I've turned the build off in Appveyor - merge as you see fit @benjeffery!

**Comment by AdminBot-tskit** [comment]
📖 Docs for this PR can be previewed [here](https://tskit-dev.github.io/msprime-docs/PR-/)

**Comment by benjeffery** [comment]
>  Docs for this PR can be previewed [here](https://tskit-dev.github.io/msprime-docs/PR-/)
Looks like some logic for adminbot is wrong. Will fix.

**Pull request merged**